transcom/mymove
# INT B-20627 Fix circleci tests and bump dependency

## [B-20627](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A982597)

## Summary

The package [otelhttp:v.0.43.0](https://pkg.go.dev/go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp@v0.43.0) is outdated and needs to be bumped up to the latest version [otelhttp:v.0.55.0](https://pkg.go.dev/go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp@v0.55.0).

Normally this is managed through [dependabot](https://github.com/dependabot) automatically but each time dependabot attempted the bump there were test failures in [circlci](https://circleci.com/?utm_source=google&utm_medium=sem&utm_campaign=sem-google-dg--uscan-en-brandAuth-tCPA-auth-brand&utm_term=g_e-circleci_c__rsa3_20210603&utm_content=sem-google-dg--uscan-en-brandAuth-tCPA-auth-brand_keyword-text_rsa-circleCI_exact-&gad_source=1&gclid=EAIaIQobChMI8eL85JCCiQMVcQGtBh33FAOREAAYASAAEgIKRPD_BwE). You can find the trail of failed PRs by going [here](https://github.com/transcom/mymove/security/dependabot/118). The [latest results](https://app.circleci.com/pipelines/github/transcom/mymove/75726/workflows/387fe08b-ca7c-404e-a457-59f37736af8f/jobs/1336255) show that the dependency bump was failing one of the automated tests run by circleci, `server_test`. 

It appears that the test is failing due to the use of deprecated methods `http.server.request_content_length` and `http.server.response_content_length`. OpenTelemetry-Go Contrib removed these in [otelhttp:v0.49.0](https://pkg.go.dev/go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp@v0.49.0) release. It appears that the new methods that need to be used are `http.server.request.size` and `http.server.response.size`.

After these tests have been fixed, the package should be able to upgraded in the[ go.mod](https://github.com/transcom/mymove/blob/main/go.mod) file to the latest version [otelhttp:v.0.55.0](https://pkg.go.dev/go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp@v0.55.0) with all circleci tests passing.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Login to GitHub.
2. Open up a **DRAFT** pull request for [MAIN-B-20627-Fix-circleci-tests-and-upgrade-dependency](https://github.com/transcom/mymove/tree/MAIN-B-20627-Fix-circleci-tests-and-upgrade-dependency) to be merged into [main](https://github.com/transcom/mymove/tree/main).
3. Scroll down to the automated tests and wait for the server_test to finish.
4. Verify the test has passed.

> [!NOTE]  
> Make sure to make the pull request a draft.

## Screenshots
![image](https://github.com/user-attachments/assets/9e0e7133-cd3a-4f5f-b8dd-caa4cd1cd61e)
